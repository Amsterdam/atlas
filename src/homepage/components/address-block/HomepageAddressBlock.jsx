import React from 'react';
import PropTypes from 'prop-types';

import HomepageBlock from '../block/HomepageBlock';

import IconAddress from '../../../../public/images/icon-adres.svg';
import IconHr from '../../../../public/images/icon-hr.svg';
import IconKadaster from '../../../../public/images/icon-kadaster.svg';

const HomepageAddressBlock = (props) => {
  const { onLinkClick } = props;

  return (
    <HomepageBlock
      onBlockLinkClick={() => onLinkClick({ dataset: 'bag', filters: {}, page: 1 })}
      title={'Adressentabel'}
      description={'Selecteer en download als spreadsheet'}
      hasTallDescription
    >
      <div className={'homepage-block__item'}>
        <button
          className={'c-link homepage-block__link'}
          title={'Bekijk Adressentabel'}
          onClick={() => onLinkClick({ dataset: 'bag', filters: {}, page: 1 })}
        >
          <span className="homepage-block__icon">
            <IconAddress />
          </span>
          <span className="homepage-block__label">
            Adressentabel
          </span>
        </button>
      </div>

      <div className={'homepage-block__item'}>
        <button
          className={'c-link homepage-block__link'}
          title={'Bekijk handelsregister-tabel'}
          onClick={() => onLinkClick({ dataset: 'hr', filters: {}, page: 1 })}
        >
          <span className="homepage-block__icon">
            <IconHr />
          </span>
          <span className="homepage-block__label">
            Handelsregister-tabel
          </span>
        </button>
      </div>

      <div className={'homepage-block__item homepage-block__item--invisible'}>
        <button
          className={'c-link homepage-block__link'}
          title={'Bekijk kadaster-tabel'}
          onClick={() => onLinkClick({ dataset: 'brk', filters: {}, page: 1 })}
        >
          <span className="homepage-block__icon">
            <IconKadaster />
          </span>
          <span className="homepage-block__label">
            Kadaster-tabel
          </span>
        </button>
      </div>

    </HomepageBlock>
  );
};

HomepageAddressBlock.propTypes = {
  onLinkClick: PropTypes.func.isRequired
};

export default HomepageAddressBlock;
