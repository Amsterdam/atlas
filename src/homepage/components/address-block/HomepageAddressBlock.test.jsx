import React from 'react';
import { shallow } from 'enzyme';

import HomepageAddressBlock from './HomepageAddressBlock';

describe('HomepageAddressBlock', () => {
  it('should render links', () => {
    const wrapper = shallow(<HomepageAddressBlock />);

    expect(wrapper).toMatchSnapshot();
  });
});
