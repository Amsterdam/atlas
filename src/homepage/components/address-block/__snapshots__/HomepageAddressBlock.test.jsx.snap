// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomepageAddressBlock should render links 1`] = `
<HomepageBlock
  blockIsLink={false}
  classes=""
  description="Selecteer en download als spreadsheet"
  hasTallDescription={true}
  linkAction={
    Object {
      "type": "atlasRouter/ADDRESSES",
    }
  }
  title="Adressentabel"
>
  <div
    className="homepage-block__item "
    key="ADRESSEN"
  >
    <Connect(Link)
      className="c-link homepage-block__link"
      title="Bekijk Adressentabel"
      to={
        Object {
          "type": "atlasRouter/ADDRESSES",
        }
      }
    >
      <span
        className="homepage-block__icon"
      >
        <Component />
      </span>
      <span
        className="homepage-block__label"
      >
        Adressentabel
      </span>
    </Connect(Link)>
  </div>
  <div
    className="homepage-block__item "
    key="HANDELSREGISTER"
  >
    <Connect(Link)
      className="c-link homepage-block__link"
      title="Bekijk handelsregister-tabel"
      to={
        Object {
          "type": "atlasRouter/ESTABLISHMENTS",
        }
      }
    >
      <span
        className="homepage-block__icon"
      >
        <Component />
      </span>
      <span
        className="homepage-block__label"
      >
        Handelsregister-tabel
      </span>
    </Connect(Link)>
  </div>
  <div
    className="homepage-block__item "
    key="KADASTER"
  >
    <Connect(Link)
      className="c-link homepage-block__link"
      title="Bekijk kadaster-tabel"
      to={
        Object {
          "type": "atlasRouter/CADASTRAL_OBJECTS",
        }
      }
    >
      <span
        className="homepage-block__icon"
      >
        <Component />
      </span>
      <span
        className="homepage-block__label"
      >
        Kadaster-tabel
      </span>
    </Connect(Link)>
  </div>
</HomepageBlock>
`;
